src/Geometry/globalgeometry.jl

```diff
@@ -146,6 +146,66 @@ function great_circle_distance(
     )
 end
 
+"""
+    euclidean_distance(pt1::XYPoint, pt2::XYPoint)
+
+Compute the 2D Euclidean distance between `pt1` and `pt2`.
+"""
+function euclidean_distance(
+    pt1::XYPoint{FT},
+    pt2::XYPoint{FT},
+) where {FT}
+    return hypot((components(pt1) .- components(pt2))...)
+end
+
+"""
+    euclidean_distance(pt1::XYPoint, pt2::XZPoint)
+
+Compute the 2D Euclidean distance between `pt1` and `pt2`.
+"""
+function euclidean_distance(
+    pt1::XYPoint{FT},
+    pt2::XZPoint{FT},
```

Comment:
Shouldn't these be the same type?

Reply:
Do we want to allow for distances between non-planar points? (i.e., one point lying on the XY plane and one that is non on the XY plane?) Am I seeing this wrong?

Reply:
Sure, but then this definition doesn't make sense: you're effectively doing:
```
sqrt((pt1.x - pt2.x)^2 + (pt1.y - pt2.z)^2)
```

Reply:
Right, I was considering the missing component as an omitted zero

Reply:
I will have to promote them then

Reply:
Would the promotion I added in [6ee763b](https://github.com/CliMA/ClimaCore.jl/pull/520/commits/6ee763b2a67fe551746c7867b57f14e07b9c5893) be an ok way to take care of that?

Reply:
Honestly I don't think we should do promotion: is there a particular case where this is needed?

Reply:
Yes, it would be needed in 2D hybrid spaces, to compute radial distance for gaussian/cosine bells types of initial conditions for advection/transport cases, similar to the 2D non-hybrid cases

Reply:
What's wrong with the promotion?

Reply:
I guess I don't quite see why you need promotion for those cases? Can you give an example of how it would be used?

My main concern is that unlike vectors, for positions there is nothing "special" about the 0 plane, so its not clear that `XPoint(2)` should be the same as `XYPoint(2,0)`.

It also adds potential for unexpected behavior: e.g. accidentally computing the distance from `XPoint(1)` to `YPoint(3)`.

Reply:
Ok probably I was seeing the hybrid case as needing two different types, but it may not be the case. In the case of, say a 2D hybrid space (1D CG x 1D FD) the coordinates field would be of XZPoint type, right? Then the radial distance would just be defined between two XZPoints and in that case we don't need a promotion like I was imaging. Similarly, for a 3D hybrid space (2D CG x 1D FD), the coordinates field would be XYZPoints and the distances can be computed among them. Sorry, I think the ides of the hybrid spaces and these different coordinate types as algebraic tuples and not geometric vectors (position vectors) confused me

Reply:
I removed it in [ab7216d](https://github.com/CliMA/ClimaCore.jl/pull/520/commits/ab7216d1dc2a07426d94845060a904fe8d4097a1). Please let me know if this is looking good and can be squashed/merged. Thanks a lot!